.github/workflows/checkmycode.yml

```diff
@@ -1,30 +1,165 @@
+  lint:
+    needs: build_and_test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Installing node_modules
+        run: npm install eslint --global
+      - name: Install Yarn
+        run: npm install -g yarn
+      - name: Run lint
+        run: yarn run eslint --init     
+  coverage:
+    name: Coverage
+    needs: 
+      - lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2.4.0
+        with:
+          node-version: 15
+      - name: Installing coverage
+        run: npm install check-code-coverage 
+      - name: Coveralls
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Upload coverage to Codecov 
+        uses: codecov/codecov-action@v1
+  duplicate_code_detection:
+    name: Check for duplicate code
+    needs: 
+      - coverage
+    runs-on: ubuntu-latest
+    steps:
+    - uses: platisd/duplicate-code-detection-tool@master
+      with:
+        directories: "test/"
+        # Only examine .h and .cpp files
+        file_extensions: "sh, js"
+        # Only report similarities above 5%
+        ignore_below: 5
+        # If a file is more than 70% similar to another, then the job fails
+        fail_above: 70
+        # If a file is more than 15% similar to another, show a warning symbol in the report
+        warn_above: 15    
+  sonar_scan:
+    name: Detect bugs, vulnerabilities and code smells
+    needs: 
+      - duplicate_code_detection
+    runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
-    - name: Setup Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v2
       with:
-        node-version: ${{ matrix.node-version }}
-        cache: 'npm'
+        # Disabling shallow clone is recommended for improving relevancy of reporting
+        fetch-depth: 0
+      # Triggering SonarQube analysis as results of it are required by Quality Gate check
+    - name: SonarQube Scan
+      uses: sonarsource/sonarqube-scan-action@master
+      env:
+        SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+    - name: SonarQube Quality Gate check
+      uses: sonarsource/sonarqube-quality-gate-action@master
+      with:
+       scanMetadataReportFile: target/sonar/report-task.txt
+      # Force to fail step after specific time
+      timeout-minutes: 5
+      env:
+       SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+  upload_to_artifactory:
+    name: Artifactory
+    needs: sonar_scan
+    runs-on: ubuntu-latest
+    steps:
+      - name: Archive staging artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist-without-markdown
+          path: |
+            dist
+            !dist/**/*.md
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v2
+        with:
+          name: code-coverage-report
+          path: output/test/code-coverage.html
```

Comment:
How do you prove that this archiving will work? That there are no mistakes in the path / filename?

Reply:
Artifactory upload was not tested as part of the current workflow. it was added as reference for future implementation.